homeassistant/components/device_tracker/icloud.py

```diff
@@ -1,100 +1,497 @@
+            ICLOUDTRACKERS[accountname].lost_iphone(devicename)
 
-        The session timeouts if we are not using it so we
-        have to re-authenticate & this will send an email.
-        """
-        api.authenticate()
-        _LOGGER.info("Authenticate against iCloud")
+    hass.services.register(DOMAIN, 'lost_iphone',
+                           lost_iphone)
 
-    seen_devices = {}
+    def update_icloud(call):
+        """Call the update function of an icloud account."""
+        accountname = call.data.get(ATTR_ACCOUNTNAME)
+        devicename = call.data.get(ATTR_DEVICENAME)
+        if accountname is None:
+            for account in ICLOUDTRACKERS:
+                ICLOUDTRACKERS[account].update_icloud(devicename)
+        elif accountname in ICLOUDTRACKERS:
+            ICLOUDTRACKERS[accountname].update_icloud(devicename)
+    hass.services.register(DOMAIN,
+                           'update_icloud', update_icloud)
 
-    def update_icloud(now):
-        """Authenticate against iCloud and scan for devices."""
-        try:
-            keep_alive(None)
-            # Loop through every device registered with the iCloud account
-            for device in api.devices:
-                status = device.status()
-                dev_id = slugify(status['name'].replace(' ', '', 99))
-
-                # An entity will not be created by see() when track=false in
-                # 'known_devices.yaml', but we need to see() it at least once
-                entity = hass.states.get(ENTITY_ID_FORMAT.format(dev_id))
-                if entity is None and dev_id in seen_devices:
-                    continue
-                seen_devices[dev_id] = True
-
-                location = device.location()
-                # If the device has a location add it. If not do nothing
-                if location:
-                    see(
-                        dev_id=dev_id,
-                        host_name=status['name'],
-                        gps=(location['latitude'], location['longitude']),
-                        battery=status['batteryLevel']*100,
-                        gps_accuracy=location['horizontalAccuracy']
-                    )
-        except PyiCloudNoDevicesException:
-            _LOGGER.info('No iCloud Devices found!')
-
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, update_icloud)
-
-    update_minutes = list(range(0, 60, config[CONF_INTERVAL]))
-    # Schedule keepalives between the updates
-    keepalive_minutes = list(x for x in range(0, 60, KEEPALIVE_INTERVAL)
-                             if x not in update_minutes)
-
-    track_utc_time_change(hass, update_icloud, second=0, minute=update_minutes)
-    track_utc_time_change(hass, keep_alive, second=0, minute=keepalive_minutes)
+    def reset_account_icloud(call):
+        """Reset an icloud account."""
+        accountname = call.data.get(ATTR_ACCOUNTNAME)
+        if accountname is None:
+            for account in ICLOUDTRACKERS:
```

Comment:
The old tracker ignored devices based on known_devices. Neither the old or new will honour any dont track configurations with this restart, since no config gets reloaded.

In the new version new devices will be picked up through reset since its not automatic anymore :+1: